app/components/SignupForm.tsx

```diff
@@ -1,35 +1,179 @@
 "use client";
-import React from "react";
 
-const SignupForm: React.FC = () => {
-  const handleSubmit = async (e: React.FormEvent) => {
+import React, { useState } from "react";
+
+const SignupForm = () => {
+  const [form, setForm] = useState({
+    id: "",
+    fullname: "",
+    email: "",
+    password: "",
+    confirm_password: "",
+  });
+
+  const [idError, setIdError] = useState("");
+  const [fullnameError, setFullnameError] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  const [confirmPasswordError, setConfirmPasswordError] = useState("");
+
+  const handleChange = (e) => {
+    setForm({ ...form, [e.target.name]: e.target.value });
+  };
+
+  const validateForm = () => {
+    let isValid = true;
+
+    if (!form.id) {
+      setIdError("값을 입력해주세요.");
+      isValid = false;
+    } else if (form.id.length < 5) {
+      setIdError("최소 5자 이상 입력해주세요.");
+      isValid = false;
+    } else if (form.id.length > 15) {
+      setIdError("최대 15자 이하로 입력해주세요.");
+      isValid = false;
+    } else {
+      setIdError("");
+    }
+
+    if (!form.fullname) {
+      setFullnameError("값을 입력해주세요.");
+      isValid = false;
+    } else {
+      setFullnameError("");
+    }
+
+    const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+    if (!emailRegex.test(form.email)) {
+      setEmailError("이메일 형식에 맞게 입력해주세요.");
+      isValid = false;
+    } else {
+      setEmailError(""); // Clear error message
+    }
+
+    const passwordRegex = /^[A-Za-z0-9]+$/;
+    if (!form.password) {
+      setPasswordError("값을 입력해주세요.");
+      isValid = false;
+    } else if (!passwordRegex.test(form.password)) {
+      setPasswordError("영문과 숫자만 입력해주세요.");
+      isValid = false;
+    } else if (form.password.length < 8) {
+      setPasswordError("최소 8자 이상 입력해주세요.");
+      isValid = false;
+    } else if (form.password.length > 20) {
+      setPasswordError("최대 20자 이하로 입력해주세요.");
+      isValid = false;
+    } else {
+      setPasswordError("");
+    }
+
+    if (!form.confirm_password) {
+      setConfirmPasswordError("값을 입력해주세요.");
+      isValid = false;
+    } else if (form.confirm_password !== form.password) {
+      setConfirmPasswordError("비밀번호가 일치하지 않습니다.");
+      isValid = false;
+    } else {
+      setConfirmPasswordError("");
+    }
+
+    return isValid;
+  };
+
+  const handleSubmit = (e) => {
```

Comment:
```suggestion
  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
```
여기도 마찬가지로 type이 빠져있습니다.